### PR TITLE
feat!: use microgenerator

### PR DESCRIPTION
**Pull request by busunkim96**
Re-opening #227 for cleaner comment history and reviewer list.

"Jump to" doesn't work because there are too many files. Here are direct links:
* [UPGRADING.md](https://github.com/googleapis/python-dialogflow/pull/239/files#diff-f411a5009131db5a6c7242b49ca78488f2d274ef9e5e5f4de9262a42108a5386)
* [setup.py](https://github.com/googleapis/python-dialogflow/pull/239/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7)
* Samples start [here](https://github.com/googleapis/python-dialogflow/pull/239/files#diff-8a449cb8cfe206673dca9b9a5aab9c42c6cb2cb138b82620a8c0af4e2e90437e)

### CODE REVIEW

File: samples/snippets/document_management.py [hunk 1]
```diff
@@ -46,19 +46,20 @@ def create_document(project_id, knowledge_base_id, display_name, mime_type,
...
+        project_id, knowledge_base_id)
 
-    document = dialogflow.types.Document(
+    document = dialogflow.Document(
         display_name=display_name, mime_type=mime_type,
         content_uri=content_uri)
 
     document.knowledge_types.append(
-        dialogflow.types.Document.KnowledgeType.Value(knowledge_type))
+        getattr(dialogflow.Document.KnowledgeType, knowledge_type)
```
Comment (busunkim96) [inline samples/snippets/document_management.py:59]: Is this an acceptable way to go from a string to the enum, or should I do something else? @software-dov 

```py
    class KnowledgeType(proto.Enum):
        r"""The knowledge type of document content."""
        KNOWLEDGE_TYPE_UNSPECIFIED = 0
        FAQ = 1
        EXTRACTIVE_QA = 2
```

**Comment by busunkim96** [comment]
@software-dov @leahecole This is now ready for review. :)

**Comment by leahecole** [comment]
I don't know too much about dialogflow, but the tests is a good sign to me, and the upgrading guide looks good!

**Review by software-dov — APPROVED** [review]
UPGRADING looks good, samples look good. I'm guessing the samples all run.

**Pull request merged**